examples/animation.rs

```diff
@@ -7,83 +7,94 @@ use std::rc::Rc;
 use winit::event::{Event, KeyEvent, WindowEvent};
 use winit::event_loop::{ControlFlow, EventLoop};
 use winit::keyboard::{Key, NamedKey};
-use winit::window::WindowBuilder;
+use winit::window::Window;
+
+include!("./utils/winit_app.rs");
 
 fn main() {
     let event_loop = EventLoop::new().unwrap();
-    let window = Rc::new(WindowBuilder::new().build(&event_loop).unwrap());
+    let start = Instant::now();
 
-    #[cfg(target_arch = "wasm32")]
-    {
-        use winit::platform::web::WindowExtWebSys;
+    let app = winit_app::WinitAppBuilder::with_init(|event_loop| {
+        let window = {
+            let window = event_loop.create_window(Window::default_attributes());
+            Rc::new(window.unwrap())
+        };
 
-        web_sys::window()
-            .unwrap()
-            .document()
-            .unwrap()
-            .body()
-            .unwrap()
-            .append_child(&window.canvas().unwrap())
-            .unwrap();
-    }
+        #[cfg(target_arch = "wasm32")]
+        {
+            use winit::platform::web::WindowExtWebSys;
 
-    let context = softbuffer::Context::new(window.clone()).unwrap();
-    let mut surface = softbuffer::Surface::new(&context, window.clone()).unwrap();
+            web_sys::window()
+                .unwrap()
+                .document()
+                .unwrap()
+                .body()
+                .unwrap()
+                .append_child(&window.canvas().unwrap())
+                .unwrap();
+        }
```

Comment:
This can be replaced with [`WindowAttributes::with_append()`](https://docs.rs/winit/0.30.0/winit/window/struct.WindowAttributes.html#method.with_append) now.